### PR TITLE
chore: remove implicit record stream assumption from builder names

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
 - Rename record stream builders to just `StreamBuilder`'s to reduce confusion in migration to block stream.

### CODE REVIEW

**Review by david-bakin-sl — APPROVED** [review]
LGTM, for smart contract service only (totally bland, a true "chore").

Surprising to me to see in one place at one time how the usage of `ContractCallRecordBuilder` is all over smart contract service.  I guess it's unavoidable - it's a key point of communication between the smart contract service and the outside world.  But still, a key dependency.  Wonder if there's any alternative that would decouple things a bit more, that would be worth it.  Sometimes you just need the dependency/coupling.  (Same of course for token service.)

File: hedera-node/hedera-token-service/src/main/java/com/hedera/node/app/service/token/records/TokenBaseRecordBuilder.java [hunk 1]
```diff
@@ -17,13 +17,13 @@
...
 import com.hedera.hapi.node.base.TokenType;
-import com.hedera.node.app.spi.workflows.record.SingleTransactionRecordBuilder;
+import com.hedera.node.app.spi.workflows.record.StreamBuilder;
 import edu.umd.cs.findbugs.annotations.NonNull;
 
 /**
  * The base interface for Token Service record builders that record operations on Tokens.
  */
-public interface TokenBaseRecordBuilder extends SingleTransactionRecordBuilder {
+public interface TokenBaseRecordBuilder extends StreamBuilder {
```
Comment (MrValioBg) [inline hedera-node/hedera-token-service/src/main/java/com/hedera/node/app/service/token/records/TokenBaseRecordBuilder.java:26]: Shouldn't this also be TokenBaseStreamBuilder?

File: hedera-node/hedera-token-service/src/main/java/com/hedera/node/app/service/token/records/TokenBurnStreamBuilder.java [hunk 2]
```diff
@@ -23,7 +23,7 @@
  * A {@code RecordBuilder} specialization for tracking the side effects of a {@code TokenBurn}
```
Comment (MrValioBg) [inline hedera-node/hedera-token-service/src/main/java/com/hedera/node/app/service/token/records/TokenBurnStreamBuilder.java:23]: Maybe we could also improve the JavaDocs for clarity, as here it's stated as "RecordBuilder."

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM!

File: hedera-node/hedera-token-service/src/main/java/com/hedera/node/app/service/token/records/TokenBurnStreamBuilder.java (hunk 2 above)
Comment (tinker-michaelj) [reply hedera-node/hedera-token-service/src/main/java/com/hedera/node/app/service/token/records/TokenBurnStreamBuilder.java]: Definitely...did a global replace of `{@code RecordBuilder}` with `{@code StreamBuilder}`...realized there are also a huge number of occurrences of the string "record builder"...will change those more opportunistically. 😓

**Review by MrValioBg — APPROVED** [review]
LGTM! Thank you @tinker-michaelj

**Review by david-bakin-sl — APPROVED** [review]
LGTM

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM! Thanks @tinker-michaelj

**Pull request merged**